File: 1972-rotating-the-box/1972-rotating-the-box.cpp
class Solution {
public:
    vector<vector<char>> rotateTheBox(vector<vector<char>>& boxGrid) {
        int row = boxGrid.size();
        int col = boxGrid[0].size();
        vector<vector<char>> box_rotated(col, vector<char>(row));

        for (int i = 0; i < row; i++){
            int cell = col - 1;
            for (int j = col - 1; j >= 0; j--){
                if (boxGrid[i][j] == '*'){
                    cell = j - 1;
                }
                else if (boxGrid[i][j] == '#') {
                    boxGrid[i][j] = '.';
                    boxGrid[i][cell--] = '#';
                }
            }
        }
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                box_rotated[j][row - i - 1] = boxGrid[i][j];
            }
        }
        return box_rotated;
        
    }
};

File: 1972-rotating-the-box/1972-rotating-the-box.java
class Solution {
    public char[][] rotateTheBox(char[][] boxGrid) {
    int m = boxGrid.length;
    int n = boxGrid[0].length;

    char [][] result = new char[n][m];
    for (int i = 0; i < m; ++i)
        for (int j = n - 1, k = n - 1; j >= 0; --j) {
            result[j][m - i - 1] = '.';
            if (boxGrid[i][j] != '.') {
                k = boxGrid[i][j] == '*' ? j : k;
                result[k--][m - i - 1] = boxGrid[i][j];
            }
        }
    return result;
}
}